jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/tools/Util.java

```diff
@@ -185,4 +186,51 @@ static void exitJVMWithReasonAndHelp(String error, String help) {
 	static boolean checkHelpOption(String option) {
 		return Arrays.stream(HELP_OPTIONS).anyMatch(option::equalsIgnoreCase);
 	}
+
+	static List<String> findMatchVMIDs(List<VirtualMachineDescriptor> vmds, String firstArg) {
+		IPC.logMessage("findMatchVMIDs firstArg = " + firstArg); //$NON-NLS-1$
+		ArrayList<String> vmids = new ArrayList<>();
+		String currentVMID = AttachHandler.getVmId();
+		try {
+			long pid = Long.parseLong(firstArg);
+			boolean includeAllVMIDs = (pid == 0);
+			// this is the virtual machine identifier which is usually an OS process ID
+			for (VirtualMachineDescriptor vmd : vmds) {
+				String vmid = vmd.id();
+				if (includeAllVMIDs
+						|| firstArg.equals(vmid)
+				) {
+					IPC.logMessage("add vmid = " + vmid); //$NON-NLS-1$
+					if (vmid.equals(currentVMID) && !AttachHandler.selfAttachAllowed) {
+						IPC.logMessage("skip itself, vmid = " + vmid); //$NON-NLS-1$
```

Comment:
I suggest uses of "itself" should be "self"; here and on line 222.

Reply:
Updated